osu.Game/Screens/Edit/Timing/TimingSection.cs

```diff
@@ -42,6 +48,15 @@ void saveChanges()
             {
                 if (!isRebinding) ChangeHandler?.SaveState();
             }
+
+            bpmTextEntry.Bindable.BindValueChanged(val =>
+            {
+                if (!Beatmap.AdjustNotesOnOffsetBPMChange.Value || ControlPoint.Value == null)
+                    return;
+
+                TimingSectionAdjustments.SetHitObjectBPM(Beatmap, ControlPoint.Value, val.OldValue);
+                Beatmap.UpdateAllHitObjects();
```

Comment:
Operation does not undo properly (try changing bpm and then pressing ctrl-z).

Review will apply to all places where these new helpers are getting called.